### PR TITLE
Add a miniature canvas to centralize mocking

### PR DESCRIPTION
**Pull request by cobbal**
[run-ipad-tests]

### CODE REVIEW

File: CanvasCore/CanvasCore/React Native Modules/NativeLoginManager.m [hunk 1]
```diff
@@ -24,28 +24,33 @@
...
-    _sharedInstance = login;
-}
-
 + (NativeLogin *)sharedInstance {
+    static dispatch_once_t onceToken;
+    static NativeLogin *_sharedInstance;
+    dispatch_once(&onceToken, ^{
+        _sharedInstance = [[NativeLogin alloc] initActual];
+    });
     return _sharedInstance;
```
Comment (cobbal) [inline CanvasCore/CanvasCore/React Native Modules/NativeLoginManager.m]: Fixes a race condition where login could happen before RN was set up

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/5ji71nf) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2F5ji71nf)
teacher: [Link](https://rebrand.ly/4vfssln) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2F4vfssln)
parent: [Link](https://rebrand.ly/zgrnkms) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fzgrnkms)

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please run `yarn update-headers`. The following do not have the correct license header: 
* TestsFoundation/TestsFoundation/MiniCanvas/LoggingHttpServer.swift
* TestsFoundation/TestsFoundation/MiniCanvas/State/MiniAssignment.swift
* TestsFoundation/TestsFoundation/MiniCanvas/State/MiniCanvasState.swift
* TestsFoundation/TestsFoundation/MiniCanvas/State/MiniCourse.swift
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: None

[MBL-14031](https://instructure.atlassian.net/browse/MBL-14031)
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/API/GradingPeriodRequestables.swift [hunk 2]
```diff
@@ -18,6 +18,7 @@
 
 import Foundation
 
+// https://canvas.instructure.com/doc/api/grading_periods.html#method.grading_periods.index
```
Comment (vanwagonet) [inline Core/Core/API/GradingPeriodRequestables.swift]: thanks for adding this.

File: TestsFoundation/TestsFoundation/MiniCanvas/MiniCanvasEndpoints.swift [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+    public static let endpoints: [MiniCanvasServer.Endpoint] = [
+        // MARK: Account Notifications
+        // https://canvas.instructure.com/doc/api/account_notifications.html
+        .apiRequest(GetAccountNotificationsRequest()) { request in request.state.accountNotifications },
+        .apiRequest(DeleteAccountNotificationRequest(id: ":id")) { request in
+            let id = ID(request[":id"]!)
+            request.state.accountNotifications.removeAll(where: { $0.id == id })
+            return APINoContent()
+        },
```
Comment (vanwagonet) [inline TestsFoundation/TestsFoundation/MiniCanvas/MiniCanvasEndpoints.swift]: I really like how you designed these routes. It's easy to understand.

**Pull request merged**